### PR TITLE
RHMAP-18726 Clone COMPONENT repositories into env.WORKSPACE

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by grdryn — COMMENTED** [review]
@vchepeli I agree that the whitespace here was kind of an inconsistent mess, but I prefer the 4 spaces (I think we've got that in most places for groovy) rather than 2.

Also, because there are whitespace changes on almost every line, it's very difficult to see what the meaningful changes are. Is it just L16, or are there other lines with actual changes?

**Comment by ghost** [comment]
@grdryn I prefer for scripts 2,2,4 and for JVM languages 4,4,8. This all files I take as groovy scripts and not groovy classes. Plus when you use GitSCM plugin its too wide when you have 4 spaces. I changed it to 4 now

File: vars/checkoutGitRepo.groovy [hunk 1]
```diff
@@ -28,17 +47,16 @@ def call(body) {
...
 
     try {
-      checkoutWithConfig(config)
+        checkoutWithConfig(config)
     } catch (Exception e) {
-        if (backupBranch) {
-          config.branch = backupBranch
-          checkoutWithConfig(config)
+        if (isBackupConfigured) {
+            config.branch = 'backupBranch'
```
Comment (grdryn) [inline vars/checkoutGitRepo.groovy]: Why does this change to a static string of `'backupBranch'`, rather than the value of `config.backupBranch`?

Comment (vchepeli) [reply vars/checkoutGitRepo.groovy]: Need to change it back :)

Comment (ghost) [reply vars/checkoutGitRepo.groovy]: just found that little bit awkward as backupBranch is String and boolean at same time :). Changed it back

Comment (grdryn) [reply vars/checkoutGitRepo.groovy]: Yeah, you're right that the 'quantum typing' of the `backupBranch` variable is not very nice. How about the best of both worlds?

```groovy
final Boolean isBackupConfigured = config.backupBranch ? true : false

// ...

if (isBackupConfigured) {
    config.branch = config.backupBranch
}
```

Comment (ghost) [reply vars/checkoutGitRepo.groovy]: @grdryn I think we can leave it as it is/Its readable but for some newbies should be confused :) and ternary operator is Java style. Let's do it Groovy ;)

**Review by grdryn — APPROVED** [review]
I'm approving this with the assumption that you've checked that it doesn't cause the release pipeline jobs or cluster lifecycle pipeline jobs to fail.

**Comment by ghost** [comment]
> I'm approving this with the assumption that you've checked that it doesn't cause the release pipeline jobs or cluster lifecycle pipeline jobs to fail.

@grdryn should I run some jobs or how I can check it?

**Comment by grdryn** [comment]
@vchepeli You can run the `configure_jenkins.sh` script from the jenkins-bob-builder repo, with the `UPDATE_TEST_JOBS` env var set to true. From your instance, you can then run the `test_cluster_lifecycle_extended` and `test_pipeline_release_extended` jobs.

Of course, also make sure that you set the pipeline library in the global Jenkins config to use the version from your PR here.

**Pull request merged**